rest/rest-api/src/main/java/org/apache/brooklyn/rest/api/ApplicationApi.java

```diff
@@ -110,15 +112,32 @@ public Response createFromYaml(
                     required = true)
             String yaml);
 
+    @Beta
+    @PUT
+    @Path("/{application}")
+    @Consumes({"application/x-yaml",
+            // see http://stackoverflow.com/questions/332129/yaml-mime-type
+            "text/yaml", "text/x-yaml", "application/yaml"})
+    @ApiOperation(
+            value = "[BETA] Create and start a new application from YAML, with the given id",
+            response = org.apache.brooklyn.rest.domain.TaskSummary.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "Undefined entity or location"),
+            @ApiResponse(code = 409, message = "Application already registered")
```

Comment:
@m4rkmckenna weird - my test `ApplicationResourceTest.testDeploymentFailsOnDuplicateAppId` asserts that it gives back a 409. I'll investigate further (building Brooklyn, and testing via the REST api directly).